### PR TITLE
add/change functions for ais-embedded

### PR DESCRIPTION
**Pull request by FelixKuhnAnsys**

### CODE REVIEW

File: pkg/externalfunctions/ansysgpt.go [hunk 1]
```diff
@@ -497,3 +498,254 @@ func AnsysGPTGetSystemPrompt(query string, prohibitedWords []string, template st
...
+//   - userTemplate: the user template for the final request
+//   - query: the user query
+//   - history: the conversation history
+//   - prohibitedWords: the list of prohibited words
+//   - errorList1: the list of error words
+//   - errorList2: the list of error words
+//
+// Returns:
+//   - stream: the stream channel
+func AisPerformLLMFinalRequest(systemTemplate string, userTemplate string, query string, history []sharedtypes.HistoricMessage, context []sharedtypes.ACSSearchResponse, prohibitedWords []string, errorList1 []string, errorList2 []string, isStream bool) (message string, stream *chan string) {
```
Comment (AlejandroFernandezLuces) [inline pkg/externalfunctions/ansysgpt.go:672]: These long lines make me think that the linter is not being passed here, we should fix that in another PR.

File: pkg/externalfunctions/ansysgpt.go [hunk 2]
```diff
@@ -497,3 +498,254 @@ func AnsysGPTGetSystemPrompt(query string, prohibitedWords []string, template st
...
+	for _, message := range history {
+		switch message.Role {
+		case "user":
+			historyMessages += "`HumanMessage`: `" + message.Content + "`\n"
+		case "assistant":
+			historyMessages += "`AIMessage`: `" + message.Content + "`\n"
+		}
+	}
+
+	// create json string fron context
```
Comment (AlejandroFernandezLuces) [inline pkg/externalfunctions/ansysgpt.go:686]:
```suggestion
	// create json string from context
```

File: pkg/externalfunctions/ansysgpt.go [hunk 3]
```diff
@@ -497,3 +498,254 @@ func AnsysGPTGetSystemPrompt(query string, prohibitedWords []string, template st
...
+	}
+
+	// Create map for the data to be used in the template
+	dataMap := make(map[string]string)
+	dataMap["query"] = query
+	dataMap["chat_history"] = historyMessages
+	dataMap["context"] = contextString
+	dataMap["prohibit_word_list"] = prohibitedWordsString
+	dataMap["error_list_1"] = errorList1String
+	dataMap["error_list_2"] = errorList2String
```
Comment (AlejandroFernandezLuces) [inline pkg/externalfunctions/ansysgpt.go:722]: Out of curiosity, why two error lists? If they are the same, I would merge them. If not, rename to make clear what each of them contains if possible.

File: pkg/functiontesting/functiontesting.go [hunk 4]
```diff
@@ -36,5 +36,5 @@ func TestAnsysGPTACSSearchIndex(indexName string, query string) {
 
 	filter := externalfunctions.AnsysGPTExtractFieldsFromQuery(query, filedValues, defaultFields)
 	output := externalfunctions.AnsysGPTACSSemanticHybridSearchs(query, embeddedQuery, indexNames, filter, 10)
-	fmt.Println(output)
+	fmt.Println(len(output))
```
Comment (AlejandroFernandezLuces) [inline pkg/functiontesting/functiontesting.go:39]: Should this print be removed?

**Review by AlejandroFernandezLuces — APPROVED** [review]
LGTM, some minor comments:

File: pkg/externalfunctions/ansysgpt.go (hunk 1 above)
Comment (FelixKuhnAnsys) [reply pkg/externalfunctions/ansysgpt.go]: linter does not seem to mind, but i changed it anyway =)

File: pkg/externalfunctions/ansysgpt.go (hunk 3 above)
Comment (FelixKuhnAnsys) [reply pkg/externalfunctions/ansysgpt.go:731]: ace wants them seperate. there is a difference in the type of error i think or something...

File: pkg/externalfunctions/ansysgpt.go (hunk 1 above)
Comment (AlejandroFernandezLuces) [reply pkg/externalfunctions/ansysgpt.go]: Thanks for changing it!

In any case, I will check the linter config, the default line length is set to 120 chars, I think this line surpasses that length and it should have been autocorrected by pre-commit

File: pkg/functiontesting/functiontesting.go (hunk 4 above)
Comment (FelixKuhnAnsys) [reply pkg/functiontesting/functiontesting.go:39]: just my testing function for the indexes. as i manually need to check the output, i think the print statement is appropriate

**Pull request merged**